### PR TITLE
Feature/fetch implementation

### PR DESCRIPTION
**Pull request by hvohr**
What I did: Implemented fetch API and remove hardcoded data files. 

Did this break anything?
- [x] No

Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x]  Bug fix (non-breaking change which fixes an issue)

Checklist:
- [x]  The code runs locally
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

What's next? Refactor HTML, refactor CSS and repair the delete functionality

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -1,4 +1,12 @@
-// Your fetch requests will live here!
+const getData = (type) => {
+  if (!['ingredients', 'users', 'recipes'].includes(type)) {
+    throw new Error("check spelling")
+  }
```
Comment (andreasorensen) [inline src/apiCalls.js:4]: I love the way you set this up!!! 
This is amazing!

File: src/apiCalls.js [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
+    return fetch(`https://what-s-cookin-starter-kit.herokuapp.com/api/v1/${type}`)
+      .then((response => response.json()))
+      .then((data) => { return data })
+}
```
Comment (Denzel-Turambi) [inline src/apiCalls.js:8]: I like this fetch request! very clean and dry!

**Review by Denzel-Turambi — APPROVED** [review]

File: src/domUpdates.js [hunk 3]
```diff
@@ -224,43 +229,51 @@ function renderFilteredRecipes() {
...
 
 const viewSelectedRecipe = event => {
+  getData('ingredients').then(({ingredients}) => {
+  getData('recipes').then(({recipes}) => {
   const recipeName = event.target.id;
-  const selectedRecipe = findRecipe(recipeData, recipeName);
-  const recipeCost = calculateRecipeCost(selectedRecipe, ingredientsData);
-  const ingredients = displayIngredients(recipeData, ingredientsData, recipeName)
+  const selectedRecipe = findRecipe(recipes, recipeName);
+  const recipeCost = calculateRecipeCost(selectedRecipe, ingredients);
```
Comment (andreasorensen) [inline src/domUpdates.js:249]: ooooh thanks for changing that from hardcoded!

**Review by andreasorensen — APPROVED** [review]
Almost there!!!

File: src/apiCalls.js [hunk 4]
```diff
@@ -1,4 +1,12 @@
-// Your fetch requests will live here!
+const getData = (type) => {
+  if (!['ingredients', 'users', 'recipes'].includes(type)) {
+    throw new Error("check spelling")
```
Comment (fernandorobles97) [inline src/apiCalls.js:3]: Great error call here!

**Review by fernandorobles97 — APPROVED** [review]

**Pull request merged**